lib/trento/infrastructure/software_updates/adapter/suma_http_executor.ex

```diff
@@ -3,44 +3,42 @@ defmodule Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor do
   SUMA Http requests executor
   """
 
-  alias Trento.Infrastructure.SoftwareUpdates.SumaApi
-
   @callback login(
               base_url :: String.t(),
               username :: String.t(),
               password :: String.t(),
-              use_ca_cert :: boolean()
+              ca_cert :: String.t() | nil
             ) ::
               {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
 
   @callback get_system_id(
               base_url :: String.t(),
               auth :: String.t(),
               fully_qualified_domain_name :: String.t(),
-              use_ca_cert :: boolean()
+              ca_cert :: String.t() | nil
             ) ::
               {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
 
   @callback get_relevant_patches(
               base_url :: String.t(),
               auth :: String.t(),
               system_id :: pos_integer(),
-              use_ca_cert :: boolean()
+              ca_cert :: String.t() | nil
             ) ::
               {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
 
   @callback get_upgradable_packages(
               base_url :: String.t(),
               auth :: String.t(),
               system_id :: pos_integer(),
-              use_ca_cert :: boolean()
+              ca_cert :: String.t() | nil
             ) ::
               {:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}
 
   @behaviour Trento.Infrastructure.SoftwareUpdates.Suma.HttpExecutor
 
   @impl true
-  def login(base_url, username, password, use_ca_cert \\ false) do
+  def login(base_url, username, password, ca_cert \\ nil) do
```

Comment:
Do we need the `\\ nil`? I mean, we can just have `ca_cert`, it doesn't really convert to some default if it's `nil` :+1: